src/Microsoft.OData.Core/UriParser/Visitors/IsCollectionTranslator.cs

```diff
@@ -14,6 +14,20 @@ namespace Microsoft.OData.UriParser
     /// </summary>
     internal sealed class IsCollectionTranslator : PathSegmentTranslator<bool>
     {
+        private static IsCollectionTranslator instance;

```

Comment:
There's no need for the field here. It's best to follow this guidance for singletons:

https://teams.microsoft.com/l/message/19:a6a2625839704b38b18625b23879d8d3@thread.tacv2/1645486366604?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=ae0a236b-39a3-42d0-9e34-19dd177ea7cf&parentMessageId=1645486366604&teamName=gdebruin%20discussion%20posts&channelName=Maintainability%20Monday&createdTime=1645486366604

I created [this](https://github.com/KenitoInc/odata.net/pull/1) PR with the changes if you'd like to take it.

Reply:
@corranrogue9 Merged your PR into this PR